firmware/boron-ins-fsm/test/consoleFunctionTests.cpp

```diff
@@ -437,4 +441,71 @@ SCENARIO("Force_Reset", "[force reset]") {
+        unsigned long initial_state3_stillness_timer_value = 30005;
+        state3_stillness_timer = initial_state3_stillness_timer_value;
+        hasDurationAlertBeenSent = true;
+        hasStillnessAlertBeenSent = false;
+
+        WHEN("the function is called with any string") {
+            int returnVal = reset_stillness_timer_for_alerting_session("any string");
+
+            THEN("The function should return the difference between old stillness timer value and now") {
+                // Testing a one-second range to prevent test flakiness
+                unsigned long expectedValue = millis() - initial_state3_stillness_timer_value;
+                REQUIRE((returnVal >= expectedValue - 1000 && returnVal <= expectedValue));
+            }
+
+            THEN("The new stillness timer value is the current value of millis()") {
+                // Testing a one-second range to prevent test flakiness
+                int currentTimeInMillis = (int)millis();
+                REQUIRE((state3_stillness_timer >= currentTimeInMillis - 1000 && state3_stillness_timer <= currentTimeInMillis));
+            }
+        }
+    }
+
+    GIVEN("A session that has already had a Stillness Alert") {
+        unsigned long initial_state3_stillness_timer_value = 30005;
+        state3_stillness_timer = initial_state3_stillness_timer_value;
+        hasDurationAlertBeenSent = false;
+        hasStillnessAlertBeenSent = true;
+
+        WHEN("the function is called with any string") {
+            int returnVal = reset_stillness_timer_for_alerting_session("any string");
+
+            THEN("The function should return the difference between old stillness timer value and now") {
+                // Testing a one-second range to prevent test flakiness
+                unsigned long expectedValue = millis() - initial_state3_stillness_timer_value;
+                REQUIRE((returnVal >= expectedValue - 1000 && returnVal <= expectedValue));
+            }
+
+            THEN("The new stillness timer value is the current value of millis()") {
+                // Testing a one-second range to prevent test flakiness
+                int currentTimeInMillis = (int)millis();
+                REQUIRE((state3_stillness_timer >= currentTimeInMillis - 1000 && state3_stillness_timer <= currentTimeInMillis));
+            }
+        }
+    }
+
+    GIVEN("A session that has not yet sent any alerts") {
+        unsigned long initial_state3_stillness_timer_value = 30005;
+        state3_stillness_timer = initial_state3_stillness_timer_value;
+        hasDurationAlertBeenSent = false;
+        hasStillnessAlertBeenSent = false;
+
+        WHEN("the function is called with any string") {
+            int returnVal = reset_stillness_timer_for_alerting_session("any string");
+
+            THEN("The function should return -1") {
+                REQUIRE(returnVal == -1);
+            }
+
+            THEN("The stillness timer value did not change") {
```

Comment:
I'd suggest opting for AND_THEN over AND in this situation because it conveys that we're dealing with two assertions within the same test structure.

Reply:
Oh cool. I didn't know about the `AND_THEN` syntax here. But looking at this documentation (https://github.com/catchorg/Catch2/blob/devel/docs/test-cases-and-sections.md#bdd-style-test-cases) , I don't think that `AND_THEN` is what I want to use. 

If I were to use `AND_THEN` then this would only run this single test:
```
GIVEN("A session that has not yet sent any alerts")
WHEN("the function is called with any string")
THEN("The function should return -1")
AND_THEN("The stillness timer value did not change")
```

but what I really want is for this to run two different tests with two different `REQUIRE` assertion:
```
GIVEN("A session that has not yet sent any alerts")
WHEN("the function is called with any string")
THEN("The function should return -1")
```
and
```
GIVEN("A session that has not yet sent any alerts")
WHEN("the function is called with any string")
THEN("The stillness timer value did not change")
```

It is generally good practice to only have one assertion per test. This is because if a test fails, then it doesn't execute the rest of the test code. So in this case, if the function **didn't** return -1, then it would fail there and wouldn't even run the code where we check to see if the stillness timer value changed or not. This gives us less information than if they were executed as two separate tests.